mm2src/lp_ordermatch.rs

```diff
@@ -672,10 +737,29 @@ async fn process_sync_pubkey_orderbook_state(
         .collect();
 
     let pair_orders_diff = try_s!(pair_orders_diff);
+    let mut protocol_infos = HashMap::new();
+    let pair_orders_diff = pair_orders_diff
+        .into_iter()
+        .map(|(pair, trie)| {
+            let new_trie = trie.map_to(|uuid, order| match order {
+                Some(o) => {
+                    protocol_infos.insert(o.uuid, BaseRelProtocolInfo {
+                        base: o.base_protocol_info.clone(),
+                        rel: o.rel_protocol_info.clone(),
+                    });
+                },
+                None => {
+                    protocol_infos.remove(uuid);
```

Comment:
I don't get why we need this remove? can't we just use
```
if let Some(o) = order {
   protocol_infos.insert(o.uuid, BaseRelProtocolInfo {
                        base: o.base_protocol_info.clone(),
                        rel: o.rel_protocol_info.clone(),
                    });
}
```

Reply:
None means that the order was removed, so I assumed we don't need its protocol info - it would allow saving a bit of network traffic. However, I now think that it's a possible bug - I need to check what happens if the delta has both the order and its removal afterward.

Reply:
Not a bug - we will use the default value temporary, and the order will be removed immediately here: 
https://github.com/KomodoPlatform/atomicDEX-API/blob/59f181521a373cb817a5b07b0de3aa4a2aab62d8/mm2src/lp_ordermatch.rs#L158-L168